### PR TITLE
[#2216] Fixed issue where share-editor was reloading unneccesarily, also...

### PR DESCRIPTION
**Pull request by dseif**
... fixed comment issue

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/1a5c9a974696fb0/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/ea8ff2a27256015/output.txt

File: src/editor/module.js [hunk 1]
```diff
@@ -49,18 +50,27 @@ define( [ "core/eventmanager", "core/trackevent", "./editor",
...
       _toggler.state = false;
 
-      if( _currentEditor ) {
+      if ( _currentEditor ) {
         _currentEditor.close();
       }
-      _currentEditor = Editor.create( editorName, butter );
-      _currentEditor.open( _editorContentArea, openData );
+
+      console.log( editorName, persist );
```
Comment (secretrobotron) [inline src/editor/module.js]: Remove please!

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/1a5c9a974696fb0/output.txt

Total script time: 1.71 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/ea8ff2a27256015/output.txt

Total script time: 1.43 mins

Server is running on http://pr946.butyr.org/templates/basic/
Run the Test Suite: http://pr946.butyr.org/test/

File: src/editor/share-editor.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
         embedHeight = embedDimensions[ 1 ],
         tooltip;
 
-    authorInput.value = butter.project.author === "Anonymous" || !butter.project.author ? "" : butter.project.author;
+    authorInput.value = !butter.project.author ? "" : butter.project.author;
```
Comment (secretrobotron) [inline src/editor/share-editor.js]: This is just less confusing if you remove the `!` from the conditional and swap the values. That doesn't change the result, right?

File: src/editor/module.js [hunk 3]
```diff
@@ -49,18 +50,27 @@ define( [ "core/eventmanager", "core/trackevent", "./editor",
      * Opens an editor corresponding to the given editor name if it exists
      *
      * @param {String} editorName: Name of editor to open
+     * @param {Booelean} persist: Indicate whether or not the editor should be recreated each time it wants to be opened
      */
-    _this.openEditor = function( editorName, forceTrayOpen, openData ) {
+    _this.openEditor = function( editorName, persist ) {
```
Comment (secretrobotron) [inline src/editor/module.js]: Changing this signature is a tiny bit dangerous. There's a bug in your patch as a result of this change that prevents you from editing TrackEvents. `editTrackEvent` uses the `openData` parameter to pass in a TrackEvent to the trackevent-editor. `forceTrayOpen` looks relatively useless now, so replacing it with `persist` is probably good enough.

File: src/editor/editor.js [hunk 4]
```diff
@@ -32,6 +32,8 @@ define( [ "util/lang", "util/xhr",
     register: function( name, layoutSrc, ctor ) {
       __editors[ name ] = {
         create: ctor,
+        // by default, likes not reload an editors content each time it's opened and closed
```
Comment (secretrobotron) [inline src/editor/editor.js]: This comment doesn't seem to make sense.

File: src/editor/editor.js [hunk 5]
```diff
@@ -32,6 +32,8 @@ define( [ "util/lang", "util/xhr",
...
+        persist: true,
```
Comment (secretrobotron) [inline src/editor/editor.js]: Why not make this an option you can pass in as a parameter to `register`?

File: src/editor/header.js [hunk 6]
```diff
@@ -26,7 +26,9 @@ define([], function(){
     }, false );
 
     _shareButton.addEventListener( "click", function( e ) {
-      editorModule.openEditor( "share-properties" );
+      editorModule.openEditor( "share-properties", {
+        persist: true
```
Comment (secretrobotron) [inline src/editor/header.js]: Do you need this anymore?

File: src/plugin/plugin-list.js [hunk 7]
```diff
@@ -59,9 +58,9 @@ define( [ "util/dragndrop", "util/lang", "editor/editor", "text!layouts/plugin-l
 
     // Open the plugin-list editor right after butter is finished starting up
     butter.listen( "ready", function() {
-      butter.editor.openEditor( "plugin-list", true );
+      butter.editor.openEditor( "plugin-list", {
+        persist: true
```
Comment (secretrobotron) [inline src/plugin/plugin-list.js]: Probably don't need this here either now. It seems to load fine with persistence enabled, which is specified in the registration of the editor.

File: src/ui/header.js [hunk 8]
```diff
@@ -65,7 +65,9 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
           _userData.showErrorDialog( "There was a problem saving your project. Please try again." );
           return;
         } else {
-          butter.editor.openEditor( "share-properties" );
+          butter.editor.openEditor( "share-properties", {
+            persist: true
```
Comment (secretrobotron) [inline src/ui/header.js]: Same as https://github.com/mozilla/butter/pull/946/files#L3R30

File: editors/googlemap-editor.js [hunk 9]
```diff
@@ -324,7 +324,5 @@
         removeMapListeners();
       }
     });
-
```
Comment (cadecairos) [inline editors/googlemap-editor.js]: These are random changes...

File: src/editor/share-editor.js [hunk 10]
```diff
@@ -31,7 +31,7 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
         embedHeight = embedDimensions[ 1 ],
         tooltip;
 
-    authorInput.value = butter.project.author === "Anonymous" || !butter.project.author ? "" : butter.project.author;
+    authorInput.value = butter.project.author ? butter.project.author : "";
```
Comment (cadecairos) [inline src/editor/share-editor.js]: wasn't this fixed in another ticket?

File: src/editor/share-editor.js [hunk 11]
```diff
@@ -106,21 +103,27 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
...
-        headerPreviewBtn.href = e.url;
+        headerPreviewBtn.href = url;
 
         updateEmbed( projectURL.value.replace( "/v/", "/e/" ) );
-        socialMedia.hotLoad( shareFacebook, socialMedia.facebook, e.url );
-        socialMedia.hotLoad( shareTwitter, socialMedia.twitter, e.url );
-        socialMedia.hotLoad( shareGoogle, socialMedia.google, e.url );
+        // if any of the buttons haven't loaded, or if we aren't logged in
+        if ( !shareFacebook.childNodes.length || !shareTwitter.childNodes.length ||
+             !shareGoogle.childNodes.length || !butter.cornfield.authenticated() ) {
```
Comment (cadecairos) [inline src/editor/share-editor.js:127]: Why hotLoad when they're _not_ logged in as opposed to when they are logged in?

File: src/editor/share-editor.js [hunk 12]
```diff
@@ -150,7 +153,7 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
         projectName.value = butter.project.name;
       }
 
-      butter.project.author = authorInput.value || "Anonymous";
+      butter.project.author = authorInput.value || "";
```
Comment (cadecairos) [inline src/editor/share-editor.js]: same as above, this should already be fixed. might merge cleanly though

File: src/editor/share-editor.js [hunk 13]
```diff
@@ -205,7 +208,9 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
         save();
       }
     });
-    butter.listen( "logout", displayLogin );
+    butter.listen( "logout", function() {
+      displayLogin();
+    });
```
Comment (cadecairos) [inline src/editor/share-editor.js]: why wrap the fn call in an anon fn?

File: editors/googlemap-editor.js (hunk 9 above)
Comment (dseif) [inline editors/googlemap-editor.js]: it seems that way now. I was touching actual content in these files in an earlier version of this patch, but didn't end up using it. If it's a big deal, I can remove all of these.

File: src/editor/share-editor.js (hunk 10 above)
Comment (dseif) [inline src/editor/share-editor.js]: Probably, guess we'll see in the rebase

File: src/editor/share-editor.js (hunk 11 above)
Comment (dseif) [inline src/editor/share-editor.js:127]: I believe we did this to combat a race condition that was being hit to ensure the buttons load properly.

File: src/editor/share-editor.js (hunk 12 above)
Comment (dseif) [inline src/editor/share-editor.js]: Yea, guess we'll see when I rebase

File: src/editor/share-editor.js (hunk 13 above)
Comment (dseif) [inline src/editor/share-editor.js]: ah this was probably due to a previous attempt as well, will revert

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/eeb744169413357/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/eeb744169413357/output.txt

Total script time: 1.89 mins

Server is running on http://pr946.butyr.org/templates/basic/
Run the Test Suite: http://pr946.butyr.org/test/

File: src/editor/share-editor.js [hunk 14]
```diff
@@ -111,21 +108,27 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
...
+        headerPreviewBtn.href = url;
 
         updateEmbed( projectURL.value.replace( "/v/", "/e/" ) );
-        socialMedia.hotLoad( shareFacebook, socialMedia.facebook, e.url );
-        socialMedia.hotLoad( shareTwitter, socialMedia.twitter, e.url );
-        socialMedia.hotLoad( shareGoogle, socialMedia.google, e.url );
+        // if any of the buttons haven't loaded, or if we aren't logged in
+        if ( !shareFacebook.childNodes.length || !shareTwitter.childNodes.length ||
+             !shareGoogle.childNodes.length || !butter.cornfield.authenticated() ) {
+          socialMedia.hotLoad( shareFacebook, socialMedia.facebook, url );
```
Comment (cadecairos) [inline src/editor/share-editor.js:128]: There's a ticket that removes this Facebook line. make sure if this lands before/after that we handle the merge conflict accordingly.

File: templates/assets/editors/image/popcorn.image.editor.js [hunk 15]
```diff
@@ -0,0 +1,242 @@
+/* This Source Code Form is subject to the terms of the MIT license
```
Comment (cadecairos) [inline templates/assets/editors/image/popcorn.image.editor.js]: Where'd this come from?

**Pull request closed without merging**